packages/core/helper-plugin/src/features/StrapiApp.tsx

```diff
@@ -0,0 +1,125 @@
+import * as React from 'react';
+
+import { TranslationMessage } from 'types';
+
+interface Permission {
+  action: string;
+  subject: string | null;
+}
+
+interface MenuItem {
+  to: string;
+  icon: React.ComponentType;
+  intlLabel: TranslationMessage;
+  permissions?: Permission[];
+  Component?: React.ComponentType;
+}
+
+/* -------------------------------------------------------------------------------------------------
+ * Context
+ * -----------------------------------------------------------------------------------------------*/
+
+interface Plugin {
+  apis: Record<string, unknown>;
+  injectionZones: Record<string, unknown>;
+  initializer: ({ setPlugin }: { setPlugin: (pluginId: string) => void }) => null | null;
+  isReady: boolean;
+  name: string;
+  pluginId: string;
+}
+
+interface StrapiAppSettingLink {
+  id: string;
+  to: string;
+  intlLabel: TranslationMessage;
+  Component: React.ComponentType;
+  permissions: Permission[] | [];
+}
+
+interface StrapiAppSetting {
+  id: string;
+  intlLabel: TranslationMessage;
+  links: StrapiAppSettingLink[];
+}
+
+interface RunHookSeries {
+  (hookName: string, asynchronous: true): Promise<unknown>;
+  (hookName: string, asynchronous: false | undefined): unknown;
+}
+
+interface RunHookWaterfall {
+  <InitialValue, Store>(
+    hookName: string,
+    initialValue: InitialValue,
+    asynchronous: false | undefined,
+    store: Store
+  ): unknown;
+  <InitialValue, Store>(
+    hookName: string,
+    initialValue: InitialValue,
+    asynchronous: true,
+    store: Store
+  ): Promise<unknown>;
+}
+
+interface StrapiAppContextValue {
+  menu: MenuItem[];
+  plugins: Record<string, Plugin>;
+  settings: Record<string, StrapiAppSetting>;
+  getPlugin: (pluginId: string) => Plugin | undefined;
+  runHookParallel: (hookName: string) => Promise<unknown>;
+  runHookWaterfall: RunHookWaterfall;
+  runHookSeries: RunHookSeries;
+}
+
+const StrapiAppContext = React.createContext<StrapiAppContextValue>({
+  getPlugin: () => undefined,
+  menu: [],
+  plugins: {},
+  runHookParallel: () => Promise.resolve(),
+  runHookWaterfall: () => Promise.resolve(),
+  runHookSeries: () => Promise.resolve(),
+  settings: {},
+});
```

Comment:
this is an interesting one, what value do you see in having these defaults set up? should we have them or should we omit them because they're not very useful and could give the sense that something works...? 🤔

Reply:
In that case we would be making them optional on the context, right? If they aren't provided the app breaks, wouldn't it be confusing to a user if the app is broken because optional properties were not provided? I believe a promise resolving undefined will break the app but probably still isn't clear why... Let me know if I'm missing something but not convinced they should be optional.

Reply:
If the app breaks because they're not provided then you're probably right your approach, maybe i'd write what you've just said above in a comment so future people know. This pattern is a bit annoying tbf, if we wanted to, we could look at implementing https://www.npmjs.com/package/@radix-ui/react-context... maybe something for when we shuffle about and move these back into the admin package.